test/test_jit.py

```diff
@@ -4641,6 +4641,35 @@ def forward(self, x):
         # shouldn't throw a type error
         torch.jit.script(MyMod())
 
+    @_tmp_donotuse_dont_inline_everything
+    def test_lazy_script(self):
+        def untraceable(x):
+            if x.ndim > 2:
+                print("hello")
+            else:
+                print("goodbye")
+            return x + 2
+
+        # Non-working example
+        def fn(x):
+            return untraceable(x)
+
+        with self.capture_stdout():
+            traced_bad = torch.jit.trace(fn, [torch.ones(2, 2)])
+
+        FileCheck().check_not("goodbye").check_not("hello").run(traced_bad.graph)
+
+        # Working example
+        untraceable = torch.jit._lazy_script_while_tracing(untraceable)
+
+        def fn2(x):
+            return untraceable(x)
+
+        with self.capture_stdout():
```

Comment:
why capture stdout? you can @suppress_warnings if warnings are being supressed ?

Reply:
there are prints in the test